model-engine/model_engine_server/common/config.py

```diff
@@ -81,9 +83,17 @@ def cache_redis_url(self) -> str:
 
         assert self.cache_redis_azure_host and infra_config().cloud_provider == "azure"
         username = os.getenv("AZURE_OBJECT_ID")
-        password = DefaultAzureCredential().get_token("https://redis.azure.com/.default").token
+        token = DefaultAzureCredential().get_token("https://redis.azure.com/.default")
+        password = token.token
+        global redis_cache_expiration
+        redis_cache_expiration = token.expires_on
         return f"rediss://{username}:{password}@{self.cache_redis_azure_host}"
 
+    @property
+    def cache_redis_url_expiration(self) -> Optional[int]:
```

Comment:
I originally thought this was something representing a "timedelta", should we call it `cache_redis_url_expiration_timestamp`? (or otherwise make it clear it represents a timestamp/point in time)